tests/test_exacttarget_pagination.py

```diff
@@ -0,0 +1,52 @@
+from base import ExactTargetBase
+import tap_tester.connections as connections
+import tap_tester.menagerie as menagerie
+import tap_tester.runner as runner
+
+class ExactTargetPagination(ExactTargetBase):
+
+    def name(self):
+        return "tap_tester_exacttarget_pagination"
+
+    def get_properties(self, *args, **kwargs):
+        props = super().get_properties(*args, **kwargs)
+        props['batch_size'] = '1'
+        return props
+
+    def test_run(self):
+        page_size = 1
+        expected_streams = self.streams_to_select()
+
+        conn_id = connections.ensure_connection(self)
+        runner.run_check_mode(self, conn_id)
+
+        found_catalogs = menagerie.get_catalogs(conn_id)
+        self.select_found_catalogs(conn_id, found_catalogs, only_streams=expected_streams)
+
+        sync_record_count = self.run_and_verify_sync(conn_id)
+
+        synced_records = runner.get_records_from_target_output()
+
+        for stream in expected_streams:
+            with self.subTest(stream=stream):
+                # expected values
+                expected_primary_keys = self.expected_primary_keys()
+
+                # collect information for assertions from sync based on expected values
+                record_count_sync = sync_record_count.get(stream, 0)
+                primary_keys_list = [(message.get('data').get(expected_pk) for expected_pk in expected_primary_keys)
```

Comment:
Instead of this use: `tuple([message['data']['expected_pk'] for expected_pk in expected_primary_keys])` as first comprehension here that iterates over the expected_primary_keys, is just making generators.

Reply:
Done